js/flightlog_parser.js

```diff
@@ -1092,7 +1092,13 @@ var FlightLogParser = function(logData) {
                  * save space for positive values). So we need to convert those very large unsigned values into their
                  * corresponding 32-bit signed values.
                  */
-                value = (value | 0) + that.sysConfig.minthrottle;
+                if((that.sysConfig.firmwareType == FIRMWARE_TYPE_BETAFLIGHT  && semver.gte(that.sysConfig.firmwareVersion, '4.0.0')) ||
+                   (that.sysConfig.firmwareType == FIRMWARE_TYPE_CLEANFLIGHT && semver.gte(that.sysConfig.firmwareVersion, '2.6.0'))) {
+                    // the value ranges from 0-1000 in the log data
+                    value = (value | 0) + 1000;
+                } else {
+                    value = (value | 0) + that.sysConfig.minthrottle;
+                }
```

Comment:
I'm not too sure about this... it works but maybe is not the correct solution. This code is under the case `case FLIGHT_LOG_FIELD_PREDICTOR_MINTHROTTLE` that implies that the value must add the minthrottle to get the correct one. 
This is defined in the firmware here: https://github.com/betaflight/betaflight/blob/9e7e263abbd7089fe4ad07595d456b6de1412780/src/main/blackbox/blackbox.c#L189-L193

Maybe is better to use another predictor. Is the rcCommand[throttle] now restricted to the 1000 to 2000 range as the others input do? It true, then maybe is better to use the same predictor than the others rcCommand.

Reply:
Yes, you're right. Probably should use another predictor. The raw data actually ranges from 0-1000 with the related firmware change. The previous incorrect version ranges from -70 to 930 (assuming the default `min_throttle = 1070`).

One advantage of using a new predictor is that I could remove the version check logic since only 4.0+ will write data with the new predictor. But the issues around using and old BBE with BF4.0+ still apply as highlighted here: 
https://github.com/betaflight/betaflight/pull/7465#issuecomment-457852644

So it's not clear if we should move forward with this. But if we do I agree that it makes sense to go the new predictor route.

Reply:
Maybe is better to range this from 1000 to 2000, in the same way than the other channels, and use the predictor 1500. In this way if I'm not wrong all will work in older versions.

Reply:
The other channels are signed +-500 while the throttle channel is unsigned so I don't see how that would work or help with older versions.

Reply:
I didn't know it was unsigned. The predictor only serves to generate the correct value. The viewer expects all channels end with a range between 1000-2000. It is indifferent the version of the firmware or the viewer. You can write a signed value (in place of the unsigned that it sends now) and the viewer will understand it if you use the correct predictor.

Reply:
No, the other rcCommand values for roll, pitch and yaw range from -500 to 500. BBE expects these values so no predictor is used and the data is sent through unmodified. Throttle ranges from 1000-2000 but for some reason the decision was made to scale it 0-1000 in the data (but erroneously using min_throttle). I think it comes from older iterations of the firmware where the rcCommand value was ranged from min_throttle to max_throttle unlike it is now where it ranges from PWM_RANGE_MIN to PWM_RANGE_MAX. All part of the differentiating inputs from outputs (min_throttle is an output). So as the firmware evolved I think this got left behind. I'm not really sure why adjusting the range downward is needed - other than it might possibly save a couple bytes. The only way we could make it work without changes to BBE is to use the `FLIGHT_LOG_FIELD_PREDICTOR_1500` predictor and subtract 1500 from the value before written to the log. But that's really no different than what's going on with using min_throttle. The actual data in the log doesn't make any sense.